scripts/pylib/twister/twisterlib/platform.py

```diff
@@ -71,20 +85,18 @@ def load(self, platform_file):
         renode = testing.get("renode", {})
         self.uart = renode.get("uart", "")
         self.resc = renode.get("resc", "")
-        # if no flash size is specified by the board, take a default of 512K
-        self.flash = data.get("flash", 512)
         self.supported = set()
-        for supp_feature in data.get("supported", []):
+        for supp_feature in vdata.get("supported", data.get("supported", [])):
             for item in supp_feature.split(":"):
                 self.supported.add(item)
 
-        self.arch = data['arch']
-        self.vendor = data.get('vendor', '')
-        self.tier = data.get("tier", -1)
-        self.type = data.get('type', "na")
-        self.simulation = data.get('simulation', "na")
-        self.simulation_exec = data.get('simulation_exec')
-        self.supported_toolchains = data.get("toolchain", [])
+        self.arch = vdata.get('arch', data.get('arch', None))
+        self.vendor = board.vendor
```

Comment:
this is the single place where `board` is used. maybe pass just `board.vendor` in the method definition? It would be less confusing  when one don't have to figure what's the difference between `data` and `board`. Or add `vendor` entry to twister.yaml so everything is in one place?

Reply:
more usage from the board object will follow.